### DISCUSSION TITLE
Kafka certificates in PEM format

### DISCUSSION THREAD

**Discussion by nabokihms** [vectordotdev/vector#16792] [General]
Just wanted to share. 

I've added the feature to specify Kafka certificates in PEM format in line with other providers.
https://github.com/vectordotdev/vector/pull/15448

However, there is a bug in `rdkafka` library that only reads a single certificate for CA, not the whole chain.
https://github.com/confluentinc/librdkafka/commit/f8830a28652532009e3f16854cb9d5004d9de06b

For now, we fixed it by building vector with librdkafka >= 2.0
https://github.com/deckhouse/deckhouse/blob/7a4e2dd5af28cd444a94029d1f51392a7d5e3147/modules/460-log-shipper/images/vector/Dockerfile

**Comment by jszwedko** [comment]
Thanks for sharing @nabokihms ! It looks like there is a PR bumping the `rdkafka-sys` to 2.0.2 of `librdkafka` so hopefully we can easily pull these changes into Vector soon. https://github.com/fede1024/rust-rdkafka/pull/553

**Comment by IngeborgRupprechtX** [comment]
Thank you for sharing!
